credential/store/src/test/java/org/wildfly/security/credential/store/impl/PropertiesCredentialStoreTest.java

```diff
@@ -0,0 +1,228 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.security.credential.store.impl;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import static org.wildfly.security.encryption.SecretKeyUtil.generateSecretKey;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.NoSuchAlgorithmException;
+import java.security.Provider;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+
+import javax.crypto.SecretKey;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.wildfly.security.credential.SecretKeyCredential;
+import org.wildfly.security.credential.store.CredentialStore;
+import org.wildfly.security.credential.store.CredentialStoreException;
+import org.wildfly.security.credential.store.WildFlyElytronCredentialStoreProvider;
+
+/**
+ * Test case to test the {@code PropertiesCredentialStore} implementation.
+ *
+ * @author <a href="mailto:darran.lofthouse@jboss.com">Darran Lofthouse</a>
+ */
+public class PropertiesCredentialStoreTest {
+
+    private static final String CREATE = "create";
+    private static final String LOCATION = "location";
+    private static final String STORE_TYPE = "PropertiesCredentialStore";
+
+    private static final Provider PROVIDER = new WildFlyElytronCredentialStoreProvider();
+    private static final Random RANDOM = new Random();
+
+    @Rule
+    public TemporaryFolder tmp = new TemporaryFolder();
+
+    @Test
+    public void testEmptyStore() throws Exception {
+        File storeFile = new File(getStoragePathForNewFile());
+        assertFalse(storeFile.exists());
+        try {
+            CredentialStore credentialStore = CredentialStore.getInstance(STORE_TYPE, () -> new Provider[] { PROVIDER } );
+            credentialStore.initialize(toConfigurationMap(storeFile.getAbsolutePath(), true));
+            assertTrue(storeFile.exists());
+
+            assertEquals("Alias Count", 0, credentialStore.getAliases().size());
+
+            credentialStore = CredentialStore.getInstance(STORE_TYPE, () -> new Provider[] { PROVIDER } );
+            credentialStore.initialize(toConfigurationMap(storeFile.getAbsolutePath(), false));
+            assertEquals("Alias Count", 0, credentialStore.getAliases().size());
+        } finally {
+            if (storeFile.exists()) {
+                storeFile.delete();
+            }
+        }
+    }
+
+    @Test
+    public void testFileDoesNotExist() throws NoSuchAlgorithmException {
```

Comment:
I suggest to add a test where the initialization fails because of a corrupted entry or bad file content.

Reply:
https://issues.redhat.com/browse/ELY-2093